src/operator/numpy/np_dot-inl.h

```diff
@@ -140,17 +141,15 @@ inline void NumpyDotForward(const nnvm::NodeAttrs& attrs,
         Kernel<scalar_mul_kernel<Req>, xpu>::Launch(
           s, out.Size(), out.dptr<DType>(), tensor, scalar);
       });
-    } else if (a_shape.ndim() == 1 || b_shape.ndim() == 1) {
+    } else if (b_shape.ndim() == 1) {
       // Case 4: a is N-D array and b is 1-D array, sum product over the last axis
       MMImpl<xpu>(ctx, a, b, out, req[0]);
     } else {
       // Case 5: a is N-D array and b is M-D array, sum product over the last axis
       //         of a and the 2nd-to-last axis of b
-      // TODO(haojin2): To be implemented...
-      if (b_shape.ndim() != 2) {
-        LOG(FATAL) << "Only support case 5 when b.ndim = 2";
-      }
-      MMImpl<xpu>(ctx, a, b, out, req[0]);
```

Comment:
May need to keep this since it's the most efficient way of handling this situation `b_shape.ndim() = 2`. The others can be directed to `TensordotImpl`.

Reply:
Fixed. Thx.